components/proxy/src/test/kotlin/com/hotels/styx/services/HealthChecksTest.kt

```diff
@@ -0,0 +1,186 @@
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.system.measureTimeMillis
+
+
+class HealthChecksTest : FeatureSpec({
+
+    feature("Probe function") {
+        scenario("Returns true when object is responsive") {
+            val staticResponse = StaticResponseHandler(200, "Hello")
+
+            urlProbe(get("/healthcheck.txt").build(), 1.seconds)
+                    .invoke(staticResponse)
+                    .toMono()
+                    .block()!!.shouldBeTrue()
+        }
+
+        scenario("Returns false when object is unresponsive for N milliseconds") {
+            val neverRespond = NeverHandler()
+
+            val duration = measureTimeMillis {
+                urlProbe(get("/healthcheck.txt").build(), 100.milliseconds)
+                        .invoke(neverRespond)
+                        .toMono()
+                        .block()!!.shouldBeFalse()
+            }
+
+            duration shouldBeGreaterThan 100
+        }
+
+        scenario("Returns false when responds with 4xx error code") {
+            val errorHandler = StaticResponseHandler(400, "Hello")
+
+            urlProbe(get("/healthcheck.txt").build(), 100.milliseconds)
+                    .invoke(errorHandler)
+                    .toMono()
+                    .block()!!.shouldBeFalse()
+        }
+
+        scenario("Returns false when responds with 5xx error code") {
+            val errorHandler = StaticResponseHandler(500, "Hello")
+
+            urlProbe(get("/healthcheck.txt").build(), 100.milliseconds)
+                    .invoke(errorHandler)
+                    .toMono()
+                    .block()!!.shouldBeFalse()
+        }
+    }
+
+    feature("State transition function") {
+        val check = healthCheckFunction(2, 3)
+
+        scenario("Transitions to Active after N consecutive positive probes") {
+            check(ObjectInactive(0), true) shouldBe ObjectInactive(1)
+            check(ObjectInactive(1), true) shouldBe ObjectActive(0)
+        }
+
+        scenario("An negative probe resets successful probes count") {
+            check(ObjectInactive(1), false) shouldBe ObjectInactive(0)
+        }
+
+        scenario("An negative probe doesn't affect successful probes in Inactive state") {
+            check(ObjectInactive(0), false) shouldBe ObjectInactive(0)
+        }
+
+        scenario("Transitions to Inactive after N consecutive negative probes") {
+            check(ObjectActive(0), false) shouldBe ObjectActive(1)
+            check(ObjectActive(1), false) shouldBe ObjectActive(2)
+            check(ObjectActive(2), false) shouldBe ObjectInactive(0)
+        }
+
+        scenario("A successful probe resets inactive count") {
+            check(ObjectActive(2), true) shouldBe ObjectActive(0)
+        }
+
+        scenario("A successful probe doesn't affect negative probes in Active state") {
+            check(ObjectActive(0), true) shouldBe ObjectActive(0)
+        }
+    }
+
+    feature("CheckOriginState") {
```

Comment:
I can't see where this is asserting/verifying anything. Is there some way it can be refactored to make it easier to understand?

Reply:
This is some old throwaway code that I forgot to remove.
(Removed now).